LIPS/lip-17.md

```diff
@@ -0,0 +1,350 @@
+
+The proposed `MEVBoostRelayWhitelist` contract is non-upgradable, and its immutable owner is intended to be initialized with the [Lido DAO Aragon Agent](https://etherscan.io/address/0x3e40D73EB977Dc6a537aF587D48316feE66E9C8c) address on mainnet upon the deployment phase.
+
+The dedicated management `manager` entity is initialized with a zero address and can be set later by the contract's owner to be able to add/remove relays. The possible candidate to bear such responsibilities without sacrificing governance security is [Easy Track](https://easytrack.lido.fi/) (in this case  [`EVMScriptExecutor (Easy Track)`](https://docs.lido.fi/deployed-contracts#easy-track) should be set as `manager`).
+
+### Relay information structure
+
+It's proposed to use the following structure:
+
+```vyper
+struct Relay:
+    uri: String[MAX_STRING_LENGTH]
+    operator: String[MAX_STRING_LENGTH]
+    is_mandatory: bool
+    description: String[MAX_STRING_LENGTH]
+```
+
+where:
+
+- `uri` is the relay's URI to fetch the data
+- `operator` is the relay title or pseudonym
+- `is_mandatory` is supposed to distinguish between optional and mandatory relays
+- `description` is designated to store any additional info (unspecified at the current stage)
+
+### Adding relay
+
+To add a single new relay, `Lido DAO Aragon Agent` or `manager` (if assigned) should call `add_relay` method, passing all necessary relay structure's params described above.
+
+### Removing relay
+
+To add a single new relay, `Lido DAO Aragon Agent` or `manager` (if assigned) should call `remove_relay` method, passing the relay's `uri`.
+
+### Updating relay
+
+The proposed flow to update the relay's info is to re-add the relay (call `remove_relay` and then `add_relay`).
+
+### Reading the current whitelisted relays
+
+Anyone is allowed to call the following methods to retrieve the current relays set:
+
+- `get_relays_amount` to check how many relays are currently whitelisted by Lido DAO
+- `get_relays` to retrieve all currently whitelisted arrays
+- `get_relay_by_uri` to retrieve the whitelisted relay details by uri
+- `get_whitelist_version` to read lastly bumped version number of the whitelist
+
+### Whitelist versioning
+
+The principal purpose of the proposed contract is to be used for generating configurations containing up-to-date whitelisted relays.
+
+To facilitate config generation process, the contract contains the previously mentioned `get_whitelist_version` view method to check whether the lastly used relays whitelist was updated or not.
+
+The version bumps on every add/remove operation and doesn't have additional semantical meaning and numbering schemes.
+
+### Events
+
+All storage modification functions emit at least single event containing all necessary data to reproduce the changes by external indexers:
```

Comment:
... -> at least _**a**_ single event
consider grammarly as usual )

Reply:
Yeah, missed this one :+1: